src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3577,4 +3686,99 @@ const std::set<int32_t> ApplicationManagerImpl::GetAppsSubscribedForWayPoints()
   return subscribed_way_points_apps_list_;
 }
 
+static hmi_apis::Common_VideoStreamingProtocol::eType ConvertVideoProtocol(
+    const char* str) {
+  if (strcmp(str, "RAW") == 0) {
+    return hmi_apis::Common_VideoStreamingProtocol::RAW;
+  } else if (strcmp(str, "RTP") == 0) {
+    return hmi_apis::Common_VideoStreamingProtocol::RTP;
+  } else if (strcmp(str, "RTSP") == 0) {
+    return hmi_apis::Common_VideoStreamingProtocol::RTSP;
+  } else if (strcmp(str, "RTMP") == 0) {
+    return hmi_apis::Common_VideoStreamingProtocol::RTMP;
+  } else if (strcmp(str, "WEBM") == 0) {
+    return hmi_apis::Common_VideoStreamingProtocol::WEBM;
+  }
+  return hmi_apis::Common_VideoStreamingProtocol::INVALID_ENUM;
+}
+
+static hmi_apis::Common_VideoStreamingCodec::eType ConvertVideoCodec(
+    const char* str) {
+  if (strcmp(str, "H264") == 0) {
+    return hmi_apis::Common_VideoStreamingCodec::H264;
+  } else if (strcmp(str, "H265") == 0) {
+    return hmi_apis::Common_VideoStreamingCodec::H265;
+  } else if (strcmp(str, "Theora") == 0) {
+    return hmi_apis::Common_VideoStreamingCodec::Theora;
+  } else if (strcmp(str, "VP8") == 0) {
+    return hmi_apis::Common_VideoStreamingCodec::VP8;
+  } else if (strcmp(str, "VP9") == 0) {
+    return hmi_apis::Common_VideoStreamingCodec::VP9;
+  }
+  return hmi_apis::Common_VideoStreamingCodec::INVALID_ENUM;
+}
+
+// static
+void ApplicationManagerImpl::ConvertVideoParamsToSO(
+    smart_objects::SmartObject& output, const BsonObject* input) {
+  if (input == NULL) {
+    return;
+  }
+  BsonObject* obj = const_cast<BsonObject*>(input);
+
+  const char* protocol = bson_object_get_string(obj, "videoProtocol");
+  if (protocol != NULL) {
+    hmi_apis::Common_VideoStreamingProtocol::eType protocol_enum =
+        ConvertVideoProtocol(protocol);
+    if (protocol_enum !=
+        hmi_apis::Common_VideoStreamingProtocol::INVALID_ENUM) {
+      output["protocol"] = protocol_enum;
+    }
+  }
+  const char* codec = bson_object_get_string(obj, "videoCodec");
+  if (codec != NULL) {
+    hmi_apis::Common_VideoStreamingCodec::eType codec_enum =
+        ConvertVideoCodec(codec);
+    if (codec_enum != hmi_apis::Common_VideoStreamingCodec::INVALID_ENUM) {
+      output["codec"] = codec_enum;
+    }
+  }
+  BsonElement* element = bson_object_get(obj, "desiredHeight");
+  if (element != NULL && element->type == TYPE_INT32) {
+    output["height"] = bson_object_get_int32(obj, "desiredHeight");
+  }
+  element = bson_object_get(obj, "desiredWidth");
+  if (element != NULL && element->type == TYPE_INT32) {
+    output["width"] = bson_object_get_int32(obj, "desiredWidth");
```

Comment:
I agree, we should likely have a file specifically for BSON object keys. I will do this since it is more tied to the `Constructed Payloads` proposal than this one.